pkg/zsio/detector/reader.go

```diff
@@ -0,0 +1,48 @@
+package detector
+
+import (
+	"bytes"
+	"errors"
+	"io"
+
+	"github.com/mccanne/zq/pkg/zsio/ndjson"
+	"github.com/mccanne/zq/pkg/zsio/raw"
+	"github.com/mccanne/zq/pkg/zsio/zjson"
+	zsonio "github.com/mccanne/zq/pkg/zsio/zson"
+	"github.com/mccanne/zq/pkg/zson"
+	"github.com/mccanne/zq/pkg/zson/resolver"
+)
+
+var ErrUnknown = errors.New("input format not recognized")
+
+func NewReader(r io.Reader, n int, t *resolver.Table) (zson.Reader, error) {
+	peeker, err := NewPeeker(r, 32*1024)
+	if err != nil {
+		return nil, err
+	}
+	b := peeker.Peek()
+	head := bytes.NewReader(b)
+	if match(zsonio.NewReader(head, resolver.NewTable())) {
```

Comment:
This will fail if your first line is over 32K.

Reply:
correct

Reply:
I suppose we can wait till phil stumbles over this :trollface:

Reply:
I agree it's a problem.

Reply:
You think I'm not watching?! 😈

Reply:
(But no, I'm not too worried about a line over 32k at this point...)